### PR TITLE
Refs [TASK][34040] Handle event List Meal Screen

### PR DESCRIPTION
**Pull request by phucsaker99**
## Related Tickets
- [#Task 34040: Handle event List Meal Screen](https://edu-redmine.sun-asterisk.vn/issues/34040)
## WHAT
- Bind Data : get data pour into recyclerview, textview
- item Click : show detail meal screen
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                textDescription.text = it.name
+                textDescriptionDetail.text = it.description
+                presenter?.getMealsByIngredient(it)
+            }
+            3 -> arguments?.getParcelable<Area>(BUNDLE_MEAL)?.let {
+                textDescription.visibility = View.GONE
+                textDescriptionDetail.visibility = View.GONE
+                viewInformationMeal.visibility = View.GONE
+                presenter?.getMealsByArea(it)
+            }
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]: .

File: app/src/main/java/com/example/yummy/ui/adapter/MealByOneAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    RecyclerView.Adapter<MealByOneViewHolder>() {
+
+    private val meals = mutableListOf<Meal>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MealByOneViewHolder {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_meal, parent, false)
+        return MealByOneViewHolder(view, onItemClicked)
+    }
+
+    override fun getItemCount(): Int = meals.size
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/adapter/MealByOneAdapter.kt]:
```suggestion
    override fun getItemCount() = meals.size
```

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    override fun setupViews() {
+        initAdapter()
+        initDialog()
+    }
+
+    override fun initData() {
+        val mealRepository = context?.let { RepositoryUtils.getMealRepository(it) }
+        presenter = mealRepository?.let { MealListPresenter(this, it) }
+        presenter?.start()
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]: Đoạn này em phải check null nhiều quá, em nên check null context 1 lần là xong mà nhỉ

Comment (phucsaker99) [reply app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]: vâng a để e edit

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            }
+        }
+    }
+
+    private fun itemMealClicked(meal: Meal) {
+
+    }
+
+    companion object {
+        const val BUNDLE_MEAL = "component"
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]:
```suggestion
        const val BUNDLE_MEAL = "BUNDLE_MEAL"
```

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    private fun initDialog() {
+        context?.let {
+            loadingDialog = LoadingDialog(it)
+        }
+    }
+
+    private fun initInputData() {
+        arguments?.getParcelable<BaseParcelable>(BUNDLE_MEAL).let {
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]:
```suggestion
        arguments?.getParcelable<BaseParcelable>(BUNDLE_MEAL).let {meal -> 
```

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                    textDescriptionDetail.text = it.description
+                    presenter?.getMealsByIngredient(it)
+                }
+                it as? Area != null -> {
+                    textDescription.visibility = View.GONE
+                    textDescriptionDetail.visibility = View.GONE
+                    viewInformationMeal.visibility = View.GONE
+                    presenter?.getMealsByArea(it)
+                }
+                else -> return
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt:86]:
```suggestion
                }
```

File: app/src/main/java/com/example/yummy/ui/meallist/MealListPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+            override fun onFail(exception: Exception?) {
+                view.showError(exception?.message.toString())
+                view.hideLoading()
+            }
+        })
+    }
+
+    override fun start() {
+
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListPresenter.kt:62]:
```suggestion
    override fun start() {
```

File: app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    override fun initData() {
+        context?.let {
+            val mealRepository = RepositoryUtils.getMealRepository(it)
+            presenter = MealListPresenter(this, mealRepository)
+        }
+        presenter?.start()
+
+
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/ui/meallist/MealListFragment.kt]: không để 2 dòng trống liên tiếp

File: app/src/main/java/com/example/yummy/data/model/BaseParcelable.kt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package com.example.yummy.data.model
+
+import android.os.Parcelable
+
+interface BaseParcelable : Parcelable
```
Comment (huytq-2422) [inline app/src/main/java/com/example/yummy/data/model/BaseParcelable.kt]: thế sao em không dùng luôn Parcelable mà phải thêm 1 interface làm gì =))

**Pull request merged**